src/include/planner/analyze_plan.h

```diff
@@ -37,19 +38,22 @@ class AnalyzePlan : public AbstractPlan {
 
   explicit AnalyzePlan(storage::DataTable *table);
 
-  explicit AnalyzePlan(std::string table_name);
+  explicit AnalyzePlan(std::string table_name,
+                       concurrency::Transaction *txn = nullptr);
```

Comment:
When is the txn going to be null? I thought that the whole point of this PR was so that we explicitly pass the txn pointer to everything?